distribution/src/assembly/binary-tarball.xml

```diff
@@ -0,0 +1,133 @@
+          xsi:schemaLocation="http://maven.apache.org/ASSEMBLY/2.0.0 http://maven.apache.org/xsd/assembly-2.0.0.xsd">
+  <id>binary-tarball</id>
+  <formats>
+    <format>tar.gz</format>
+  </formats>
+  <includeBaseDirectory>true</includeBaseDirectory>
+
+  <moduleSets>
+
+    <!-- Module java-driver-core and its dependencies -->
+    <moduleSet>
+      <useAllReactorProjects>true</useAllReactorProjects>
+      <includes>
+        <include>com.datastax.oss:java-driver-core</include>
+      </includes>
+      <binaries>
+        <outputDirectory>lib/core</outputDirectory>
+        <unpack>false</unpack>
+        <dependencySets>
+          <dependencySet>
+            <outputDirectory>lib/core</outputDirectory>
+            <excludes>
+              <!--
+                For some reason, we need to exclude all other modules here, even though our
+                moduleSet targets core only
+              -->
+              <exclude>com.datastax.oss:java-driver-query-builder</exclude>
+            </excludes>
+            <useTransitiveFiltering>true</useTransitiveFiltering>
+          </dependencySet>
+        </dependencySets>
+      </binaries>
+    </moduleSet>
+
+    <!-- Module java-driver-query-builder and its dependencies -->
+    <moduleSet>
+      <useAllReactorProjects>true</useAllReactorProjects>
+      <includes>
+        <include>com.datastax.oss:java-driver-query-builder</include>
+      </includes>
+      <binaries>
+        <outputDirectory>lib/query-builder</outputDirectory>
+        <unpack>false</unpack>
+        <dependencySets>
+          <dependencySet>
+            <excludes>
+              <exclude>com.datastax.oss:java-driver-core</exclude>
+              <!-- Don't reinclude dependencies that core also has -->
+              <exclude>com.datastax.oss:java-driver-shaded-guava</exclude>
+              <exclude>com.github.stephenc.jcip:jcip-annotations</exclude>
+              <!-- TODO exclude modules other than core when they get added -->
+            </excludes>
+            <useTransitiveFiltering>true</useTransitiveFiltering>
+          </dependencySet>
+        </dependencySets>
+      </binaries>
+    </moduleSet>
+
+    <!--sources for all modules-->
+    <moduleSet>
+      <useAllReactorProjects>true</useAllReactorProjects>
+      <includes>
+        <include>com.datastax.oss:java-driver-core</include>
+        <include>com.datastax.oss:java-driver-query-builder</include>
+      </includes>
+      <binaries>
+        <unpack>false</unpack>
+        <attachmentClassifier>sources</attachmentClassifier>
+        <outputFileNameMapping>${module.artifactId}-${module.version}-src.zip
+        </outputFileNameMapping>
+        <outputDirectory>src</outputDirectory>
+        <excludes>
+          <exclude>*</exclude>
+        </excludes>
+      </binaries>
+    </moduleSet>
+
+  </moduleSets>
+
+  <fileSets>
+
+    <fileSet>
+      <directory>target/apidocs</directory>
+      <outputDirectory>apidocs</outputDirectory>
```

Comment:
it doesn't look like `apidocs` doesn't end up in the tar.gz file from what I can tell.  I see it in the target directory, but not in the tar file.

Reply:
Maybe it has something to do with how i'm running the build?  `mvn clean install -DskipTests -Prelease`

Reply:
That should work, I'll take a look at it.